### DISCUSSION TITLE
Accessing the fitted HF transformer

### DISCUSSION THREAD

**Discussion by didriksg** [autogluon/autogluon#2005] [Q&A]
After fitting a `TextPredictor` as a part of a `TabularPredictor`, is there a convenient way to access the fitted HF transformer? Accessing the transformer would be convenient for encoding sentences for other tasks. I have found a way of accessing it, but it is quite ugly and guaranteed not the intended way to access it:

```python
model = TabularPredictor.load(checkpoint_path)
model.persist_models()
hf_transformer= model._trainer.models['TextPredictor'].model._predictor._model.model
```

I have checked out the saved model generated after a fitting, but I cannot seem to find a way of loading the HF transformer by itself. Is there any existing functionality to do this, and if not, would this be something that would be of interest to others?

**Comment by sxjscience** [comment]
Hi @didriksg , are you interested in dumping the HF transformer for other usage? We can support this feature.

**Comment by didriksg** [comment]
Hi @sxjscience 
Yes, I want to use the trained HF transformer for other, separate tasks. Currently, I do this as shown in the original post, where I also must create a deepcopy of the HF transformer, as it seems that the transformer model is unloaded when it's not used, and simply having a reference to it as shown in the original post creates some problems after a while in runtime.

It would be very convenient to have this as a separate, supported feature! Dumping the HF transformer to enable loading it in a more convenient way than I do now would be a great addition! An even greater feature would be to be able to access the HF transformer directly from a loaded model (either `Tabular` or `MultiModal`) so that you do not need to have two identical copies of the HF transformerr loaded in memory simultaneously (as I do now)
